src/Microsoft.OData.Core/Json/BufferingJsonReader.cs

```diff
@@ -999,6 +1350,589 @@ private void RemoveFirstNodeInBuffer()
+
+            try
+            {
+                // Read the next node in the current reader mode (buffering or non-buffering)
+                bool result = await this.ReadInternalAsync()
+                    .ConfigureAwait(false);
+
+                if (this.asyncInnerReader.NodeType == JsonNodeType.StartObject)
+                {
+                    // If we find a StartObject node we have to read ahead and check whether this
+                    // JSON object represents an in-stream error. If we are currently in buffering
+                    // mode remember the current position in the buffer; otherwise start buffering.
+                    bool wasBuffering = this.isBuffering;
+                    BufferedNode storedPosition = null;
+                    if (this.isBuffering)
+                    {
+                        storedPosition = this.currentBufferedNode;
+                    }
+                    else
+                    {
+                        await this.StartBufferingAsync()
+                            .ConfigureAwait(false);
+                    }
+
+                    await this.ProcessObjectValueAsync()
+                        .ConfigureAwait(false);
+
+                    // Reset the reader state to non-buffering or to the previously
+                    // backed up position in the buffer.
+                    if (wasBuffering)
+                    {
+                        this.currentBufferedNode = storedPosition;
+                    }
+                    else
+                    {
+                        this.StopBuffering();
+                    }
+                }
+
+                return result;
+            }
+            finally
+            {
+                this.parsingInStreamError = false;
+            }
+        }
+
+        /// <summary>
+        /// Try to read an error structure from the stream. Return null if no error structure can be read.
+        /// </summary>
+        /// <param name="error">An <see cref="ODataError"/> instance that was read from the reader or null if none could be read.</param>
+        /// <returns>A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains a tuple comprising of:
+        /// 1). A value of true if an <see cref="ODataError"/> instance that was read; otherwise false.
+        /// 2). An <see cref="ODataError"/> instance that was read from the reader or null if none could be read.</returns>
+        private async Task<Tuple<bool, ODataError>> TryReadInStreamErrorPropertyValueAsync()
+        {
+            Debug.Assert(this.parsingInStreamError, "this.parsingInStreamError");
+            this.AssertBuffering();
+            this.AssertAsynchronous();
+
+            // We expect a start-object node here
+            if (this.currentBufferedNode.NodeType != JsonNodeType.StartObject)
+            {
+                return Tuple.Create(false, (ODataError)null);

```

Comment:
```suggestion
                return (false, (ODataError)null);

```
You could also name the fields:
`return (hasError: false, error: (ODataError)null);`

Reply:
See my comment here https://github.com/OData/odata.net/pull/2264#discussion_r759368356